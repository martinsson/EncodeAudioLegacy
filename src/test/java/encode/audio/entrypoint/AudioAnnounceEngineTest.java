package encode.audio.entrypoint;

import org.approvaltests.legacycode.LegacyApprovals;
import org.junit.Test;

import flux.AudioAnnounceTmlg;
import flux.IFluxTmlg;

public class AudioAnnounceEngineTest {

	private static final String HOST = "http://somehost/";
	private static final String REMOTE_AUDIO_FILE_NAME = "10.151.156.180Mon_Nov_04_140724_CET_2013343.wav";
	private static final String REMOTE_AUDIO_FILE_NAME2 = "10.151.156.180Tue_Nov_05_141112_CET_2013343.mp3";
	private static final String REMOTE_AUDIO_FILE_NAME3 = "10.151.156.180Tue_Nov_05_141112_CET_2013343.ogg";
	private static final String REMOTE_AUDIO_FILE_NAME4 = "10.151.156.180Tue_Nov_05_141112_CET_2013343.tmp";

	@Test
	public void coverageAudioAnnounceEngineLockdown() throws Exception {
		// Given
		String[] url = { HOST + "10.151.156.180Mon_Nov_04_140724_CET_2013343",	HOST + "10.151.156.180Tue_Nov_05_141112_CET_2013343" };
		String[] targetFormat = { "mp3", "wav", "ogg", "zip" };
		String[] remoteAudioFileName = { REMOTE_AUDIO_FILE_NAME, REMOTE_AUDIO_FILE_NAME2, REMOTE_AUDIO_FILE_NAME3, REMOTE_AUDIO_FILE_NAME4 };

		// When
		// Then
		LegacyApprovals.LockDown(this, "coverageAudioAnnounceEngine", url,	targetFormat, remoteAudioFileName);
	}

	public String coverageAudioAnnounceEngine(String url, String targetFormat, String remoteAudioFileName) {
		AudioAnnounceTmlg audioFileMessage = new AudioAnnounceTmlg(url,	targetFormat, remoteAudioFileName);
		DataObject configAudioTmp = new AudioDataObject("." + targetFormat);
		DataObject httpDataObj = new HttpDataObj("./src/test/resources/", "http://localhost/get");

		LocalHTTPSServer localServerFolder = new LocalHTTPSServer();
		LocalTmpFolder localTmpFolder = new LocalTmpFolder();
		AudioAnnounceEngine audioAnnounceEngine = new AudioAnnounceEngine(localServerFolder, localTmpFolder);

		IFluxTmlg availableEncodedAudioFile;
		try {
			availableEncodedAudioFile = audioAnnounceEngine.publishAudioFile(audioFileMessage, configAudioTmp, httpDataObj);
		} catch (AppTechnicalException e) {
			return e.getMessage();
		}
		return availableEncodedAudioFile.toString();
	}
}
