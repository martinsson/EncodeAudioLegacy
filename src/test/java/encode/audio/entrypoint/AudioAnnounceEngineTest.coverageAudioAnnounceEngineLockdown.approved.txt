[http://somehost/10.151.156.180Mon_Nov_04_140724_CET_2013343, mp3, 10.151.156.180Mon_Nov_04_140724_CET_2013343.wav] = FluxTmlg [body=AnnounceAudioTmlg [url=http://localhost/get10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3, format=mp3, fileName=10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3]] 
[http://somehost/10.151.156.180Tue_Nov_05_141112_CET_2013343, mp3, 10.151.156.180Mon_Nov_04_140724_CET_2013343.wav] = FluxTmlg [body=AnnounceAudioTmlg [url=http://localhost/get10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3, format=mp3, fileName=10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3]] 
[http://somehost/10.151.156.180Mon_Nov_04_140724_CET_2013343, wav, 10.151.156.180Mon_Nov_04_140724_CET_2013343.wav] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Tue_Nov_05_141112_CET_2013343, wav, 10.151.156.180Mon_Nov_04_140724_CET_2013343.wav] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Mon_Nov_04_140724_CET_2013343, ogg, 10.151.156.180Mon_Nov_04_140724_CET_2013343.wav] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Tue_Nov_05_141112_CET_2013343, ogg, 10.151.156.180Mon_Nov_04_140724_CET_2013343.wav] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Mon_Nov_04_140724_CET_2013343, zip, 10.151.156.180Mon_Nov_04_140724_CET_2013343.wav] = FluxTmlg [body=AnnounceAudioTmlg [url=http://localhost/get10.151.156.180Mon_Nov_04_140724_CET_2013343.zip, format=zip, fileName=10.151.156.180Mon_Nov_04_140724_CET_2013343.zip]] 
[http://somehost/10.151.156.180Tue_Nov_05_141112_CET_2013343, zip, 10.151.156.180Mon_Nov_04_140724_CET_2013343.wav] = FluxTmlg [body=AnnounceAudioTmlg [url=http://localhost/get10.151.156.180Mon_Nov_04_140724_CET_2013343.zip, format=zip, fileName=10.151.156.180Mon_Nov_04_140724_CET_2013343.zip]] 
[http://somehost/10.151.156.180Mon_Nov_04_140724_CET_2013343, mp3, 10.151.156.180Tue_Nov_05_141112_CET_2013343.mp3] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Tue_Nov_05_141112_CET_2013343, mp3, 10.151.156.180Tue_Nov_05_141112_CET_2013343.mp3] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Mon_Nov_04_140724_CET_2013343, wav, 10.151.156.180Tue_Nov_05_141112_CET_2013343.mp3] = FluxTmlg [body=AnnounceAudioTmlg [url=http://localhost/get10.151.156.180Tue_Nov_05_141112_CET_2013343.wav, format=wav, fileName=10.151.156.180Tue_Nov_05_141112_CET_2013343.wav]] 
[http://somehost/10.151.156.180Tue_Nov_05_141112_CET_2013343, wav, 10.151.156.180Tue_Nov_05_141112_CET_2013343.mp3] = FluxTmlg [body=AnnounceAudioTmlg [url=http://localhost/get10.151.156.180Tue_Nov_05_141112_CET_2013343.wav, format=wav, fileName=10.151.156.180Tue_Nov_05_141112_CET_2013343.wav]] 
[http://somehost/10.151.156.180Mon_Nov_04_140724_CET_2013343, ogg, 10.151.156.180Tue_Nov_05_141112_CET_2013343.mp3] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Tue_Nov_05_141112_CET_2013343, ogg, 10.151.156.180Tue_Nov_05_141112_CET_2013343.mp3] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Mon_Nov_04_140724_CET_2013343, zip, 10.151.156.180Tue_Nov_05_141112_CET_2013343.mp3] = FluxTmlg [body=AnnounceAudioTmlg [url=http://localhost/get10.151.156.180Tue_Nov_05_141112_CET_2013343.zip, format=zip, fileName=10.151.156.180Tue_Nov_05_141112_CET_2013343.zip]] 
[http://somehost/10.151.156.180Tue_Nov_05_141112_CET_2013343, zip, 10.151.156.180Tue_Nov_05_141112_CET_2013343.mp3] = FluxTmlg [body=AnnounceAudioTmlg [url=http://localhost/get10.151.156.180Tue_Nov_05_141112_CET_2013343.zip, format=zip, fileName=10.151.156.180Tue_Nov_05_141112_CET_2013343.zip]] 
[http://somehost/10.151.156.180Mon_Nov_04_140724_CET_2013343, mp3, 10.151.156.180Tue_Nov_05_141112_CET_2013343.ogg] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Tue_Nov_05_141112_CET_2013343, mp3, 10.151.156.180Tue_Nov_05_141112_CET_2013343.ogg] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Mon_Nov_04_140724_CET_2013343, wav, 10.151.156.180Tue_Nov_05_141112_CET_2013343.ogg] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Tue_Nov_05_141112_CET_2013343, wav, 10.151.156.180Tue_Nov_05_141112_CET_2013343.ogg] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Mon_Nov_04_140724_CET_2013343, ogg, 10.151.156.180Tue_Nov_05_141112_CET_2013343.ogg] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Tue_Nov_05_141112_CET_2013343, ogg, 10.151.156.180Tue_Nov_05_141112_CET_2013343.ogg] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Mon_Nov_04_140724_CET_2013343, zip, 10.151.156.180Tue_Nov_05_141112_CET_2013343.ogg] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Tue_Nov_05_141112_CET_2013343, zip, 10.151.156.180Tue_Nov_05_141112_CET_2013343.ogg] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Mon_Nov_04_140724_CET_2013343, mp3, 10.151.156.180Tue_Nov_05_141112_CET_2013343.tmp] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Tue_Nov_05_141112_CET_2013343, mp3, 10.151.156.180Tue_Nov_05_141112_CET_2013343.tmp] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Mon_Nov_04_140724_CET_2013343, wav, 10.151.156.180Tue_Nov_05_141112_CET_2013343.tmp] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Tue_Nov_05_141112_CET_2013343, wav, 10.151.156.180Tue_Nov_05_141112_CET_2013343.tmp] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Mon_Nov_04_140724_CET_2013343, ogg, 10.151.156.180Tue_Nov_05_141112_CET_2013343.tmp] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Tue_Nov_05_141112_CET_2013343, ogg, 10.151.156.180Tue_Nov_05_141112_CET_2013343.tmp] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Mon_Nov_04_140724_CET_2013343, zip, 10.151.156.180Tue_Nov_05_141112_CET_2013343.tmp] = encode.audio.utils.CoreException: Cant encode file! 
[http://somehost/10.151.156.180Tue_Nov_05_141112_CET_2013343, zip, 10.151.156.180Tue_Nov_05_141112_CET_2013343.tmp] = encode.audio.utils.CoreException: Cant encode file! 
